model/validation.go

```diff
@@ -75,7 +75,7 @@ func validateScripts(roleManifest *RoleManifest) validation.ErrorList {
 		usedScripts["scripts/"+relpath] = false
 		return nil
 	})
-	if err != nil && !roleManifest.validationOptions.AllowMissingScripts {
+	if err != nil && !os.IsNotExist(err) && !roleManifest.validationOptions.AllowMissingScripts {
```

Comment:
I suppose there are cases where this could lead to misleading validation errors, e.g. if the directory can't be listed because of permission errors but the validation complains about the files not being there at all.
I think it would be cleaner to handle the error but it will ony be helpful in very theoretical cases so I'm also fine with removing the code. Your call :)

Reply:
I would prefer to remove the check.

If you want to keep it, then I would like an explanation why the error is ignored when `AllowMissingScripts` is set! How does that related to errors while walking the directory?

Or you can change the check to just `if err != nil && !os.IsNotExist(err) { }`

Reply:
Alright, I removed the `!roleManifest.validationOptions.AllowMissingScripts` part of the condition.